### ISSUE TITLE
Report fields not available for output reports filename

### ISSUE THREAD

**Issue by felix-caboff** [GhostManager/Ghostwriter#350]
**Is your feature request related to a problem? Please describe.**
I am required to format the filename of the output reports in a particular way and this includes the project name.  At the moment this field is not available.  Instead only the following are:

![image](https://github.com/GhostManager/Ghostwriter/assets/125881693/17f0c064-b185-4c60-961e-12ecd43a74d6)

**Describe the solution you'd like**
It would be ideal if all the variables that are available in the report are also available in the filename within the global report options.  This gives the greatest level of flexibility.  There may also be mileage is attempting to rationalise the naming convention too.

**Comment by chrismaddalena** [comment]
Hey @felix-caboff, the project name displayed by Ghostwriter is the client name + the assessment type so you can assemble that in the report title configuration. Do you mean you want to include the codename?

When you say you want all variables available, do you mean the values under the `project` key available in the templates?

<img width="130" alt="image" src="https://github.com/GhostManager/Ghostwriter/assets/10526228/fc704a34-0fb7-4b91-9de5-5e53b9015881">

**Comment by felix-caboff** [comment]
Yes please on the whole of the project key.

**Comment by er4z0r** [comment]
I second @felix-caboff. Having access to the reporting context when constructing the filename would be very helpful.

One thing I have wondered though is whether it would be better located with the template (if only as an optional field that can be used to override the global) as opposed to the overall installation. In the end the filename is something that mostly affects the report as a final product so one could argue that the structure of the filename would better be placed with the other information that shapes the report (i.e., the template).

**Comment by chrismaddalena** [comment]
@er4z0r That's something that may be possible with via the new custom fields feature coming in v4.1 (See PR #346). We're looking at reports now, but we could potentially extend custom fields to templates too. Then a check like "if a field named `report_filename` exists, use that instead of the global value" could be a thing.

Allowing global options to be overridden by templates has been in the backlog for a long time.

I don't know if _all_ report context can be made accessible as a filename, especially once custom fields are in the mix, but we'll see about expanding what is available. Many of the values from the `project` key are very doable.

**Comment by felix-caboff** [comment]
This looks like this may have been completed in recent release: https://github.com/GhostManager/Ghostwriter/releases/tag/v4.2.0

**Comment by felix-caboff** [comment]
We have not yet upgraded (busy couple of months!).  Can anyone confirm that the above is now resolved in the above release.  I read the release note of "Added support for using Jinja2 in the report filename template configured under the Global Report Configuration inside the admin panel" as being exactly what this issue requested.

**Comment by chrismaddalena** [comment]
Yes, report filename templates have full use of Jinja2 templating now. This applies to the global report filenames and the newer filename overrides for individual templates.

**Comment by felix-caboff** [comment]
Awesome!  Closing...
